src/util/gas-factory-helpers.ts

```diff
@@ -543,3 +544,161 @@ export function initSwapRouteFromExisting(
     portionAmount: swapRoute.portionAmount,
   };
 }
+
+export const calculateL1GasFeesHelper = async (
+  route: RouteWithValidQuote[],
+  chainId: ChainId,
+  usdPool: Pair | Pool,
+  quoteToken: Token,
+  nativePool: Pair | Pool | null,
+  l2GasData?: ArbitrumGasData | OptimismGasData
+): Promise<{
+  gasUsedL1: BigNumber;
+  gasUsedL1OnL2: BigNumber;
+  gasCostL1USD: CurrencyAmount;
+  gasCostL1QuoteToken: CurrencyAmount;
+}> => {
+  const swapOptions: SwapOptionsUniversalRouter = {
+    type: SwapType.UNIVERSAL_ROUTER,
+    recipient: '0x0000000000000000000000000000000000000001',
+    deadlineOrPreviousBlockhash: 100,
+    slippageTolerance: new Percent(5, 10_000),
+  };
+  let l1Used = BigNumber.from(0);
+  let l1FeeInWei = BigNumber.from(0);
+  let gasUsedL1OnL2 = BigNumber.from(0);
+  const opStackChains = [
+    ChainId.OPTIMISM,
+    ChainId.OPTIMISM_GOERLI,
+    ChainId.OPTIMISM_SEPOLIA,
+    ChainId.BASE,
+    ChainId.BASE_GOERLI,
+  ];
+  if (opStackChains.includes(chainId)) {
+    [l1Used, l1FeeInWei] = calculateOptimismToL1SecurityFee(
+      route,
+      swapOptions,
+      l2GasData as OptimismGasData,
+      chainId
+    );
+  } else if (
+    chainId == ChainId.ARBITRUM_ONE ||
+    chainId == ChainId.ARBITRUM_GOERLI
+  ) {
+    [l1Used, l1FeeInWei, gasUsedL1OnL2] =
+      calculateArbitrumToL1SecurityFee(
+        route,
+        swapOptions,
+        l2GasData as ArbitrumGasData,
+        chainId
+      );
+  }
+
+  // wrap fee to native currency
+  const nativeCurrency = WRAPPED_NATIVE_CURRENCY[chainId];
+  const costNativeCurrency = CurrencyAmount.fromRawAmount(
+    nativeCurrency,
+    l1FeeInWei.toString()
+  );
+
+  // convert fee into usd
+  const nativeTokenPrice =
+    usdPool.token0.address == nativeCurrency.address
+      ? usdPool.token0Price
+      : usdPool.token1Price;
+
+  const gasCostL1USD: CurrencyAmount =
+    nativeTokenPrice.quote(costNativeCurrency);
+
+  let gasCostL1QuoteToken = costNativeCurrency;
+  // if the inputted token is not in the native currency, quote a native/quote token pool to get the gas cost in terms of the quote token
+  if (!quoteToken.equals(nativeCurrency)) {
+    if (!nativePool) {
+      log.info(
+        'Could not find a pool to convert the cost into the quote token'
+      );
+      gasCostL1QuoteToken = CurrencyAmount.fromRawAmount(quoteToken, 0);
+    } else {
+      const nativeTokenPrice =
+        nativePool.token0.address == nativeCurrency.address
+          ? nativePool.token0Price
+          : nativePool.token1Price;
+      gasCostL1QuoteToken = nativeTokenPrice.quote(costNativeCurrency);
+    }
+  }
```

Comment:
Can you use the helpers `getQuoteThroughNativePool` that we use elsewhere for this? It'll handle token0/1 and quote across a supplied pool